src/js/NavigraphNavdataInterface.ts

```diff
@@ -0,0 +1,294 @@
+
+export class NavigraphNavdataInterface {
+  private readonly listener: CommBusListener
+  private queue: CommBusMessage[] = []
+  private eventListeners: Callback[] = []
+  private onReadyCallback: (() => void) | null = null
+
+  private isInitialized = false
+
+  /**
+   * Creates a new NavigraphNavdataInterface
+   *
+   * @remarks
+   * `RegisterCommBusListener` is called during construction. This means that the class must be instantiated once the function is available.
+   */
+  constructor() {
+    this.listener = RegisterCommBusListener(() => {
+      this.onRegister()
+    })
+  }
+
+  /**
+   * Executes a SQL query on the active database
+   *
+   * @remarks
+   * The query must be valid SQL and must be a SELECT query. The query must not specify a special result format.
+   *
+   * @param sql - SQL query to execute
+   * @returns A promise that resolves with the result of the query
+   */
+  public async execute_sql<T>(sql: string, params: string[]): Promise<T[]> {
+    return await this.callWasmFunction("ExecuteSQLQuery", { sql, params })
+  }
+
+  /**
+   * Downloads the navdata from the given URL to the given path
+   *
+   * @param url - A valid signed URL to download the navdata from
+   * @param path - The path to download the navdata to
+   * @returns A promise that resolves when the download is complete
+   */
+  public async download_navdata(url: string, path: string): Promise<void> {
+    return await this.callWasmFunction("DownloadNavdata", { url, path })
+  }
+
+  /**
+   * Sets the download options for all future downloads
+   *
+   * @param batchSize - The number of files to delete or unzip each update (default: 10). This is a performance optimization to avoid blocking the main thread for too long.
+   * @returns A promise that resolves when the function is complete
+   */
+  public async set_download_options(batch_size: number): Promise<void> {
+    return await this.callWasmFunction("SetDownloadOptions", batch_size)
+  }
+
+  /**
+   * Sets the active DFD database to the one at the given path
+   *
+   * @remarks
+   * The path must be a valid path to a folder that contains a DFD navdata database.
+   *
+   * @param path - The path to the folder that contains the DFD navdata
+   * @returns A promise that resolves when the function is complete
+   */
+  public async set_active_database(path: string): Promise<void> {
+    return await this.callWasmFunction("SetActiveDatabase", { path })
+  }
+
+  public async get_database_info(ident: string): Promise<DatabaseInfo> {
+    return await this.callWasmFunction("GetDatabaseInfo", { ident })
+  }
+
+  public async get_airport(ident: string): Promise<Airport> {
+    return await this.callWasmFunction("GetAirport", { ident })
+  }
+
+  public async get_waypoints(ident: string): Promise<Waypoint[]> {
+    return await this.callWasmFunction("GetWaypoints", { ident })
+  }
+
+  public async get_vhf_navaids(ident: string): Promise<Waypoint[]> {
```

Comment:
Sould the return type be `Waypoint`? I see there is a `VhfNavaid` type?

Reply:
Typo! thanks for spotting that